processes/WIP-bug_bounty_proposal.md

```diff
@@ -0,0 +1,73 @@
+
+A bug bounty program is a mechanism to formally engage security researchers and developers in a way that provides benefits to the community along multiple vectors:
+
+1. Improved Security Depth of Coverage • Finding security defects is specialized work that benefits from deep expertise and fresh eyes
+1. Validate Nodejs as a secure platform • Expanding the community into the security world is a growth area and enables new audiences for nodejs to be proposed
+1. Improved Security Scope of Coverage • Reduced Risk we have bugs that are undiscovered currently exposing ourselves and our users to unnecessary risk
+1. Improved Process in SDLC • Ongoing analysis of discoveries provides an opportunity to improve security through practices and/or security focus and tooling.
+
+## Proposal Process
+
+The proposed process for considering a Bug Bounty program makes heavy reference to existing work that the Security WG has already started (establishing a triage team, organized disclosure system, etc.) and ongoing partnership with HackerOne. High level process:
+
+-[ ] Security WG reviews readiness and desire to have such a program (presume yes unless someone objects)
+-[ ] Security WG presents proposal to TSC (see next steps on how to get there)
+-[ ] TSC / Security WG bring funding proposal before Board
+-[ ] Security WG & TSC establish Bug Bounty Leader & Bug Bounty Team (will take additional coordination across org)
+
+## Next steps
+
+1. Core team readiness review and architectural decisions
+
+   - goals/kpi (scope, responsiveness goals, impact of defects, team size, budget)
+   - private vs. public
+   - existing security analysis
+   - backlog validation
+
+1. Engage other committies to let them know of the proposal (community committee as there is a PR element, others?)
+1. Review and evolve draft note into a proposal
+1. Create draft Security Page updates that describe the rules of the road
+
+## Readiness review and questions for input
+
+As preparation, the Security WG already has an organization within HackerOne and the general framework as well as the technical tools to offer a Bug Bounty program.
+
+1. Do we have a security assessment of the current node.js core? *no formal assessment, should not block, but increases risk of public bounty program*
+1. Do we have any scope limitations on what we would declare? *yes, we do, just what are they exactly?*
+1. Should we bounty "LTS" version bugs and non-LTS the same *(gut is yes)*? Release vs. HEAD *(gut is no)*?
+1. Do we have enough volunteer staff (need to get HackerOne help to know what the right number might should be)
+1. Would a private bounty program be acceptable to start? *Lots of indications from those who have gone before that we should start a closed / invite program first*
+1. Does the security-wg or another group own the communication plan for describing results to the broader community?
```

Comment:
Starting with a private one does sound like a good idea so that we can gain a better understanding of what done one requires.

Reply:
I like the idea of inviting researchers at least as a visibility thing before broad attention and PR outside of the community. I do see other comments below though about not working well for an open source project. Would like to discuss on our next meeting as it's not obvious to me why it should not work for an open source project.

Reply:
My feedback was more from just seeing other open source projects go through similar things. However, there's nothing preventing you from starting off in private. You just might not get that many reports, depending on who gets invited. If you do want to go the private route, get a list of every person who has ever submitted a report to security@nodejs.org and invite them to the program.

In any case, VDP and BBP are different things. Could still move security@nodejs.org over to HackerOne as a public vulnerability disclosure program. Separately, then start a private bug bounty program until you feel ready to go public with that aspect. Note that IBB can still go ahead and start paying bounties for things post-fix for a VDP, as that's separate.